### PR TITLE
Delete environments from an application

### PR DESCRIPTION
**Pull request by drummond-dds**
* [Pivotal Tracker](https://www.pivotaltracker.com/story/show/165156771)

```cucumber
Scenario: 
Given I am logged into AT-AT as a user with permissions to delete environments,
When I click to Application Settings,
Then I see a "Delete Environment" button next to each environment,
Then I click that button,
Then I see a warning alert saying that the action is permanent but that logs will be maintained,
And there is a text box,
And the Delete Environment button is disabled until I type "Delete" into the text box (case insensitive),
Then I click Delete Environmnet
Then the environment is (soft) deleted,
And the environment are deleted within the CSP (whatever that entails...)
Then I am sent to the application settings page,
And I see a success banner telling me that {environment name} has been deleted.
```

### CODE REVIEW

File: templates/fragments/applications/edit_environments.html [hunk 1]
```diff
@@ -122,6 +124,42 @@ <h4>{{ role }}</h4>
...
+                    {{ "common.delete_confirm" | translate }}
+                  </span>
+                </label>
+                <input id="deleted-text" v-model="deleteText">
+              </div>
+              <div class="action-group">
+                <form method="POST" action="{{ url_for('applications.delete_environment', environment_id=env['id']) }}">
+                  {{ form.csrf_token }}
+                  <button class="usa-button button-danger" v-bind:disabled="!valid">
+                    {{ "portfolios.applications.delete.button" | translate }}
```
Comment (dandds) [inline templates/fragments/applications/edit_environments.html]: This reads "delete application". If we put them in a common macro, we could pass a `resource` argument to use for the modal name and do `{{ "common.delete" | translate }} {{ resource }}` here.

File: templates/fragments/applications/edit_environments.html [hunk 2]
```diff
@@ -122,6 +124,42 @@ <h4>{{ role }}</h4>
...
+              <div class="usa-input">
+                <label for="deleted-text">
+                  <span class="usa-input__help">
+                    {{ "common.delete_confirm" | translate }}
+                  </span>
+                </label>
+                <input id="deleted-text" v-model="deleteText">
+              </div>
+              <div class="action-group">
+                <form method="POST" action="{{ url_for('applications.delete_environment', environment_id=env['id']) }}">
```
Comment (dandds) [inline templates/fragments/applications/edit_environments.html]: This is always deleting the last environment on the list for me.

File: atst/routes/applications/settings.py [hunk 3]
```diff
@@ -179,3 +179,21 @@ def delete(application_id):
             "applications.portfolio_applications", portfolio_id=application.portfolio_id
         )
     )
+
+
+@applications_bp.route("/environments/<environment_id>/delete", methods=["POST"])
+@user_can(Permissions.DELETE_ENVIRONMENT, message="delete delete_environment")
```
[the remark this answers is not in the document]
Comment (drummond-dds) [reply atst/routes/applications/settings.py]: Fixed in c1acdd81c9939d17b9fc146d6f6c40a3ad0d7d59

File: templates/fragments/applications/edit_environments.html (hunk 2 above)
Comment (drummond-dds) [reply templates/fragments/applications/edit_environments.html]: Fixed in 4fe42ceebb87c8eecea18399743ad3fa2206964d

File: templates/fragments/applications/edit_environments.html [hunk 4]
```diff
@@ -122,6 +124,42 @@ <h4>{{ role }}</h4>
...
+                    {{ "portfolios.applications.delete.button" | translate }}
+                  </button>
+                </form>
+                <div class="action-group">
+                  <a v-on:click="deleteText = ''; $root.closeModal('delete-application')" class="action-group__action icon-link icon-link--default">{{ "common.cancel" | translate }}</a>
+                </div>
+              </div>
+            </div>
+          </delete-confirmation>
+        {% endcall %}
```
[the remark this answers is not in the document]
Comment (drummond-dds) [reply templates/fragments/applications/edit_environments.html]: ef23b1d776d465cbb1959fceb46fb2e14b329940

File: templates/fragments/applications/edit_environments.html (hunk 1 above)
Comment (drummond-dds) [reply templates/fragments/applications/edit_environments.html]: f314e9238dc70c622311aa8235af19e85dccf5c3

**Comment by drummond-dds** [comment]
@dandds made PR changes

**Pull request merged**